src/Swarm/Game/Step.hs

```diff
@@ -1017,7 +1026,7 @@ addSeedBot e (minT, maxT) loc ts =
 -- | Interpret the execution (or evaluation) of a constant application
 --   to some values.
 execConst ::
-  (HasRobotStepState sig m, Has (Lift IO) sig m) =>
+  (HasRobotStepState sig m, Has TimeEffect sig m, Has (Lift IO) sig m) =>
```

Comment:
I'm glad there are some cases (e.g. `doGrab`) where we were able to simply replace the `Lift IO` effect with `TimeEffect`.  I assume we still need the `Lift IO` in the other cases?